src/pages/index.tsx

```diff
@@ -90,9 +90,18 @@ export async function getServerSideProps(context: any) {
       /Android|BlackBerry|iPhone|iPad|iPod|Opera Mini|IEMobile|WPDesktop/i
     )
   );
+
+  //Filter out duplicates, put "enkaz (wreckage)" on top
+  const reasonsPresent = new Set<string>();
+  let filteredReasons = reasons.filter((reason) => {
+    const filterCondition = !reasonsPresent.has(reason);
+    reasonsPresent.add(reason);
+    return filterCondition;
+  });
```

Comment:
I think you should either use `new Set<string>([...reasons])` or just filter without using a set. Using the filter method defeats the whole purpose of using sets.